### PR TITLE
Add documentation for configurable hierarchy

### PR DESCRIPTION
**Pull request by garethbowen**
medic/medic#3639

### CODE REVIEW

File: configuration/app-settings.md [hunk 1]
```diff
@@ -169,3 +169,30 @@ Under the `patient_reports` key in app_settings, we can setup actions to take fo
...
+
+|Property|Description|Required|
+|-------|---------|----------|
+| `id` | String identifier for the type. At some times this will be used to sort the contacts in the UI so it is recommended to using a number prefix with gaps between numbers, eg: `10-district`, `20-region`, etc. | Yes. |
+| `name_key` | The translation key used for the title for the contact profile. | No, defaults to 'contact.profile'. |
+| `group_key` | The translation key used for the title of a list of contacts of this type. | Yes. |
+| `create_key` | The translation key used on the button for creating new contacts of this type. | Yes. |
+| `edit_key` | The translation key used on the button for editing contacts of this type. | Yes. |
+| `primary_contact_key` | The translation key used to identify a person as the primary contact of contacts of this type. | No, defaults to 'Primary contact'. |
+| `parents` | An array of strings of IDs of parent types. If more than one then this type can appear in different places in the hierarchy. If more than one type lists the same type as a parent then a user will get a dropdown of places to create. | No, defaults to no parents. |
```
Comment (kennsippell) [inline configuration/app-settings.md]: Would be useful to specify the behaviour when empty.

File: configuration/app-settings.md [hunk 2]
```diff
@@ -192,4 +192,31 @@ Replications are defined under the `replications` key as an array of replication
...
+| `name_key` | The translation key used for the title for the contact profile. | No, defaults to 'contact.profile'. |
+| `group_key` | The translation key used for the title of a list of contacts of this type. | Yes. |
+| `create_key` | The translation key used on the button for creating new contacts of this type. | Yes. |
+| `edit_key` | The translation key used on the button for editing contacts of this type. | Yes. |
+| `primary_contact_key` | The translation key used to identify a person as the primary contact of contacts of this type. | No, defaults to 'Primary contact'. |
+| `parents` | An array of strings of IDs of parent types. If more than one then this type can appear in different places in the hierarchy. If more than one type lists the same type as a parent then a user will get a dropdown of places to create. If no parents then contacts of this type will be at the top of the hierarchy and cannot be added as a child of any contact. | No, defaults to no parents. |
```
Comment (kennsippell) [inline configuration/app-settings.md]: Is it supported to have a type which can be at the "top of hierarchy" and also be under another contact type (eg. clinic)? If so, how is this expressed? `parent: [undefined, "clinic"]`?

Comment (garethbowen) [reply configuration/app-settings.md]: That is not currently supported. Can you think of a use case where this would be desirable?

Comment (kennsippell) [reply configuration/app-settings.md]: No idea. Just checking for the parsing in `move-contacts`

**Pull request merged**